pkg/kubelet/logs/container_log_manager.go

```diff
@@ -176,19 +176,107 @@ func (c *containerLogManager) Start() {
 	}, logMonitorPeriod)
 }
 
+type containerWithLog struct {
+	ID        string
+	CreatedAt int64
+	logPath   string
+	State     runtimeapi.ContainerState
+}
+
+func (c *containerLogManager) addContainerToMap(container *runtimeapi.Container, path string, containersByName map[string][]containerWithLog) {
+	if container.Metadata != nil {
+		index := -1
+		for i := range containersByName[container.Metadata.Name] {
+			if containersByName[container.Metadata.Name][i].ID == container.Id {
+				index = i
+				containersByName[container.Metadata.Name][i].ID = container.Id
+				containersByName[container.Metadata.Name][i].State = container.State
+				containersByName[container.Metadata.Name][i].CreatedAt = container.CreatedAt
+				break
+			}
+		}
+		if index < 0 {
+			withLog := containerWithLog{
+				ID:        container.Id,
+				State:     container.State,
+				CreatedAt: container.CreatedAt,
+				logPath:   path,
+			}
+			containersByName[container.Metadata.Name] = append(containersByName[container.Metadata.Name], withLog)
+		} else if containersByName[container.Metadata.Name][index].logPath == "" {
+			containersByName[container.Metadata.Name][index].logPath = path
+		}
+	}
+}
+
+func (c *containerLogManager) pruneDeadContainerLogs(containersByName map[string][]containerWithLog) {
```

Comment:
is there a reason why we need to entirely re-write the log pruning code?  Based on the description, I would think this just requires extending removeExcessLogs to handle terminated containers?

Reply:
Current log rotation is based on :
```
		id := container.GetId()
```
The requirement for this PR is different: we want to keep cap on the containers (of the same pod) whose log files are to be kept.
Therefore here is some new code.

I tried to reuse existing code which turned out not to be as clean as the current formation.

I am open to suggestion on further simplification of the PR.